File: willow-api/api/src/main/groovy/ish/oncourse/willow/service/CourseClassesApi.java
package ish.oncourse.willow.service;

import ish.oncourse.willow.model.web.CourseClass;
import ish.oncourse.willow.model.web.CourseClassesParams;

import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import java.util.List;

@Path("/")
public interface CourseClassesApi  {

    @POST
    @Path("/classes")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @CollegeInfo
    List<CourseClass> getCourseClasses(CourseClassesParams courseClassesParams);
}

File: react/src/js/http/CourseClassesApi.ts
import {HttpService} from "../common/services/HttpService";
import { CommonError } from "../model/common/CommonError";
import { CourseClass } from "../model/web/CourseClass";
import { CourseClassesParams } from "../model/web/CourseClassesParams";

export class CourseClassesApi {
  constructor(private http: HttpService) {
  }

  getCourseClasses(courseClassesParams: CourseClassesParams): Promise<CourseClass[]> {
    return this.http.POST(`/classes`, courseClassesParams)
  }
}
